src/main/java/com/faforever/server/ladder/DivisionService.java

```diff
@@ -0,0 +1,144 @@
+package com.faforever.server.ladder;
+
+import com.faforever.server.config.ServerProperties;
+import com.faforever.server.entity.Division;
+import com.faforever.server.entity.Player;
+import com.faforever.server.entity.PlayerDivisionInfo;
+import com.google.common.collect.ImmutableList;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.text.MessageFormat;
+import java.util.Objects;
+
+@Service
+@Slf4j
+public class DivisionService {
+  private final ServerProperties properties;
+  private final DivisionRepository divisionRepository;
+  private final PlayerDivisionInfoRepository playerDivisionInfoRepository;
+
+  @Getter
+  private ImmutableList<Division> divisions;
+
+  @Inject
+  public DivisionService(ServerProperties properties, DivisionRepository divisionRepository, PlayerDivisionInfoRepository playerDivisionInfoRepository) {
+    this.properties = properties;
+    this.divisionRepository = divisionRepository;
+    this.playerDivisionInfoRepository = playerDivisionInfoRepository;
+  }
+
+  @PostConstruct
+  public void init() {
+    divisions = ImmutableList.copyOf(divisionRepository.findAllByOrderByLeagueAscThresholdAsc());
+  }
+
+  public Division getCurrentPlayerDivision(Player player) {
+    int season = properties.getLadder().getSeason();
```

Comment:
`NullPointerException` since `getSeason()` is currently nullable.

Reply:
I'm moving to `int`